File: app/src/main/java/mx/edu/upqroo/kristenandroid/adapters/ScheduleItemAdapter.java
package mx.edu.upqroo.kristenandroid.adapters;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import mx.edu.upqroo.kristenandroid.R;
import mx.edu.upqroo.kristenandroid.data.models.ScheduleSubject;

public class ScheduleItemAdapter extends RecyclerView.Adapter<ScheduleItemAdapter.ViewHolderSchedule>{
    private List<ScheduleSubject> mDaysList;

    public ScheduleItemAdapter(List<ScheduleSubject> daysList) {
        mDaysList = daysList;
    }

    @NonNull
    @Override
    public ViewHolderSchedule onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_schedule,parent,false);
        return new ViewHolderSchedule(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolderSchedule holder, int position) {
        ScheduleSubject scheduleSubject = mDaysList.get(position);
        holder._dayOfWeek.setText(scheduleSubject.getDay().getName());

        holder.mAdapter = new SubjectItemAdapter(scheduleSubject.getSubjects());
        holder.mRecycler.setAdapter(holder.mAdapter);
    }

    public void setData(List<ScheduleSubject> data) {
        mDaysList = data;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        return mDaysList.size();
    }

    class ViewHolderSchedule extends RecyclerView.ViewHolder {

        private TextView _dayOfWeek;
        private RecyclerView mRecycler;
        private SubjectItemAdapter mAdapter;

        ViewHolderSchedule(View itemView) {
            super(itemView);
            _dayOfWeek = itemView.findViewById(R.id.titSche);
            mRecycler = itemView.findViewById(R.id.recycler_subject);
            mRecycler.setLayoutManager(new LinearLayoutManager(itemView.getContext()));
        }
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/adapters/ScheduleItemAdapter.kt
package mx.edu.upqroo.kristenandroid.adapters


import android.os.Build
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import mx.edu.upqroo.kristenandroid.R
import mx.edu.upqroo.kristenandroid.data.models.ScheduleSubject

class ScheduleItemAdapter(private var mDaysList: List<ScheduleSubject>?)
    : RecyclerView.Adapter<ScheduleItemAdapter.ViewHolderSchedule>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderSchedule {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_schedule, parent, false)
        return ViewHolderSchedule(view)
    }

    override fun onBindViewHolder(holder: ViewHolderSchedule, position: Int) {
        val scheduleSubject = mDaysList!![position]
        holder.dayOfWeek.text = scheduleSubject.day.name

        holder.mAdapter = SubjectItemAdapter(scheduleSubject.subjects)
        holder.mRecycler.adapter = holder.mAdapter
    }

    fun setData(data: List<ScheduleSubject>) {
        mDaysList = data
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return mDaysList!!.size
    }

    inner class ViewHolderSchedule(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val dayOfWeek: TextView = itemView.findViewById(R.id.titSche)
        val mRecycler: RecyclerView = itemView.findViewById(R.id.recycler_subject)
        var mAdapter: SubjectItemAdapter? = null

        init {
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT_WATCH) {
                mRecycler.layoutManager = object : LinearLayoutManager(itemView.context,
                        VERTICAL, false) {
                    override fun canScrollHorizontally(): Boolean {
                        return false
                    }

                    override fun canScrollVertically(): Boolean {
                        return false
                    }
                }
            } else {
                mRecycler.layoutManager = LinearLayoutManager(itemView.context)
            }

        }
    }
}
